src/main/java/edu/kis/powp/jobs2d/events/SelectTestFigureOptionListener.java

```diff
@@ -20,20 +30,36 @@ public SelectTestFigureOptionListener(DriverManager driverManager, TestShape sha
 
 	@Override
 	public void actionPerformed(ActionEvent e) {
+		Job2dDriver driver = driverManager.getCurrentDriver();
+		List<DriverCommand> commands = new ArrayList<DriverCommand>();
+		ComplexCommand complexCommand = new ComplexCommand(commands);
 		switch (this.testShape) {
 			case FIGURE1:
-				FiguresJoe.figureScript1(driverManager.getCurrentDriver());
+				FiguresJoe.figureScript1(driver);
 				break;
 			case FIGURE2:
-				FiguresJoe.figureScript2(driverManager.getCurrentDriver());
+				FiguresJoe.figureScript2(driver);
 				break;
-
 			case RECTANGLE:
-				ComplexCommandFactory.getRectangle().execute(driverManager.getCurrentDriver());
+				int x = 100;
+				int y = 100;
+				int width = 100;
+				int height = 100;
+				commands.add(new SetPositionCommand(x, y));
+				commands.add(new OperateToCommand(x+width, y));
+				commands.add(new OperateToCommand(x+width, y+height));
+				commands.add(new OperateToCommand(x, y+height));
+				commands.add(new OperateToCommand(x, y));
+				complexCommand.execute(driver);
 				break;
-
 			case STAR:
-				ComplexCommandFactory.getStar().execute(driverManager.getCurrentDriver());
+				commands.add(new SetPositionCommand(60, 20));
+				commands.add(new OperateToCommand(100, 110));
+				commands.add(new OperateToCommand(10, 50));
+				commands.add(new OperateToCommand(110, 50));
+				commands.add(new OperateToCommand(20, 110));
+				commands.add(new OperateToCommand(60, 20));
+				complexCommand.execute(driver);
 				break;
```

Comment:
I asked you to change Star and Rectangle tests to LoadStarCommand and LoadRectangleCommand (SetCurrentCommand without explicit execute).